testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java

```diff
@@ -107,6 +110,14 @@ private DistributedQueryRunner(
     {
         requireNonNull(defaultSession, "defaultSession is null");
 
+        if (backupCoordinatorProperties.isPresent()) {
+            requireNonNull(coordinatorProperties, "coordinatorProperties cannot be null, if backupCoordinatorProperties is not null!");
+            requireNonNull(coordinatorProperties.get("http-server.http.port"), "the http-server.http.port of coordinatorProperties cannot be null, if backupCoordinatorProperties is not null!");
+            requireNonNull(backupCoordinatorProperties.get().get("http-server.http.port"), "the http-server.http.port of backupCoordinatorProperties cannot be null, if backupCoordinatorProperties is not null!");
+            Assertions.assertNotEquals(coordinatorProperties.get("http-server.http.port"), backupCoordinatorProperties.get().get("http-server.http.port"));
```

Comment:
Why do we need this. I would very much prefer that we do not need to set explicit listen ports and use random ones. With `8080` used by tests it is very probable to get a port conflict on test run

Reply:
> Why do we need this. I would very much prefer that we do not need to set explicit listen ports and use random ones. With `8080` used by tests it is very probable to get a port conflict on test run

I change them to random ones.